0000-encryption-api/README.md

```diff
@@ -0,0 +1,135 @@
+- Feature Name: (mal000001, encryption-api)
+- Start Date: 6-Feb-2019
+- RFC PR: (leave this empty)
+- Ursa Issue: (leave this empty)
+- Version: 1
+
+# Summary
+[summary]: #summary
+
+Encryption is considered a primary use for cryptography which Ursa does not
+currently provide. There is demand for providing easy to use encryption/decryption
+interfaces in Ursa. Ursa already has similar interfaces for signing/verification.
+The implemented algorithms provide programmers choices that aim
+to avoid bad parameter interactions. Keys are passed around as data blobs
+allowing consumers to protect them as they see fit.
+
+# Motivation
+[motivation]: #motivation
+
+Hyperledger projects will need to be able to encrypt/decrypt data using
+industry standards. To minimize mistakes and allow other projects good
+encryption schemes, Ursa will provide known good implementations with
+good APIs that are simple enough to avoid confusion.
+
+# Guide-level explanation
+[guide-level-explanation]: #guide-level-explanation
+
+Ursa should distinguish between two methods for encryption: one for asymmetric algorithms
+like RSA, and one for symmetric like AES. However the interfaces will be very similar.
+
+[Libsodium](https://libsodium.gitbook.io/doc/) for example uses these two types as public-key encryption and secret key encryption.
+
+Public key encryption is generally applied for key exchange or secret key wrapping. It requires two different keys–*public* and *private*, linked together with a one way mathematical
```

Comment:
You might want to clarify that the one way relationship that public key is derived from private key. But is it always true that you can derive public key from private key (its in Elgamal and RSA)?

Reply:
It doesn't have to be the case that public and private keys are linked together in any way.  This sentence should be taken out.

If we want to have some kind of key derivation function, that should probably be included as its own interface (but could be included in the encryption library).

Reply:
For simple encryption this is the case correct? More advanced encryption where this is not the case should be a separate RFC.

Reply:
The case where the secret key and public key are tied together is in fact a case of a more advanced protocol.

In the simplest schemes, the keys are independent.

Reply:
I agree that the language requiring the public and private keys be linked should be removed.